biogtr/models/embedding.py

```diff
@@ -34,42 +139,15 @@ def _torch_int_div(
         """
         return torch.div(tensor1, tensor2, rounding_mode="floor")
 
-    def _sine_box_embedding(
-        self,
-        boxes,
-        features: int = 512,
-        temperature: int = 10000,
-        scale: float = None,
-        normalize: bool = False,
-        **kwargs,
-    ) -> torch.Tensor:
+    def _sine_box_embedding(self, boxes: torch.Tensor) -> torch.Tensor:
         """Compute sine positional embeddings for boxes using given parameters.
 
         Args:
             boxes: the input boxes.
-            features: number of position features to use.
-            temperature: frequency factor to control spread of pos embed values.
-                A higher temp (e.g 10000) gives a larger spread of values
-            scale: A scale factor to use if normalizing
-            normalize: Whether to normalize the input before computing embedding
 
         Returns:
             torch.Tensor, the sine positional embeddings.
         """
-        # update default parameters with kwargs if available
-        params = {
-            "features": features,
-            "temperature": temperature,
-            "scale": scale,
-            "normalize": normalize,
-            **kwargs,
-        }
-
-        self.features = params["features"]
-        self.temperature = params["temperature"]
-        self.scale = params["scale"]
-        self.normalize = params["normalize"]
-
         if self.scale is not None and self.normalize is False:
             raise ValueError("normalize should be True if scale is passed")
         if self.scale is None:
```

Comment:
Prefer to set the default value of `self.scale` in a single place (either here or in the constructor)